PR: https://github.com/odoo/odoo/pull/153099

From: eccb937393ff3fb22e3d341137325782d1faf9cd
From: Florian Charlier
Date: 2024-02-22 20:32:54

Breaking data model changes score: 2, change matches:
-    _inherit = 'onboarding.onboarding'
-    _inherit = 'onboarding.onboarding.step'

Total Changes: 36

[REM] calendar: remove calendar onboarding

Banners are usually distracting and do not provide the help we expected.

Task-3724601

closes odoo/odoo#153099

Related: odoo/enterprise#56074
Related: odoo/upgrade#5657
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/calendar/models/__init__.py
+++ b/addons/calendar/models/__init__.py
@@ -14,5 +14,3 @@ from . import mail_activity
 from . import mail_activity_mixin
 from . import mail_activity_type
 from . import res_users
-from . import onboarding_onboarding
-from . import onboarding_onboarding_step

--- a/addons/calendar/models/onboarding_onboarding.py
+++ b/None
@@ -1,22 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models
-
-
-class Onboarding(models.Model):
-    _inherit = 'onboarding.onboarding'
-
-    @api.model
-    def action_close_calendar_onboarding(self):
-        self.action_close_panel('calendar.onboarding_onboarding_calendar')
-
-    def _prepare_rendering_values(self):
-        """Compute existence of invoices for company."""
-        self.ensure_one()
-        if self == self.env.ref('calendar.onboarding_onboarding_calendar', raise_if_not_found=False):
-            step = self.env.ref('calendar.onboarding_onboarding_step_setup_calendar_integration', raise_if_not_found=False)
-            if step and step.current_step_state == 'not_done':
-                credentials = self.env['res.users'].check_calendar_credentials()
-                if any(credentials[service] for service in credentials):
-                    step.action_set_just_done()
-        return super()._prepare_rendering_values()

--- a/addons/calendar/models/onboarding_onboarding_step.py
+++ b/None
@@ -1,12 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models
-
-
-class OnboardingStep(models.Model):
-    _inherit = 'onboarding.onboarding.step'
-
-    @api.model
-    def action_view_start_calendar_sync(self):
-        action = self.env["ir.actions.actions"]._for_xml_id("calendar.action_view_start_calendar_sync")
-        return action
